src/promptflow-tools/promptflow/tools/openai_gpt4v.py

```diff
@@ -11,25 +15,28 @@
 class OpenAI(ToolProvider):
     def __init__(self, connection: OpenAIConnection):
         super().__init__()
-        self.connection = connection
-        self._connection_dict = dict(self.connection)
+        self._client = OpenAIClient(
+            api_key=connection.api_key,
```

Comment:
Prefer to separate backward compatible mapping logic to be more clear and leave space for future connection name change.

Reply:
use mapping function in common.py